src/api/integrator/mapbox-gl.js

```diff
@@ -50,6 +50,17 @@ class MGLIntegrator {
                 this._layers.map(layer => layer.initCallback());
             }
             layer.paintCallback();
+            this._paintedLayers++;
+
+            if (this._paintedLayers % this._layers.length == 0) {
+                // Last layer has been painted
+                const isAnimated = this._layers.some(layer =>
+                    layer.getStyle() && layer.getStyle().isAnimated());
+                if (!isAnimated && this.map.repaint) {
+                    this.map.repaint = false;
```

Comment:
this._paintedLayers = 0 here?

Reply:
Not needed, this is the last layer, therefore (this._paintedLayers % numLayers == 0 already), and, on top of that, MGL would still end this repaint anyway.

In any case, this makes me think about multiple layer cases...

Reply:
After giving it a minute, I think it's ok with multiple layers

Reply:
But is there any issue if the number of layers is not a multiple of the max size of _paintedLayers (regarding overflow) ?

Reply:
If `_paintedLayers` overflows probably all the logic falls down. However, assuming one layer and 60FPS it will overflow after 9007199254740991/60 seconds which is more than 4 million years :smile:

Reply:
We can live with it :joy: